job-server/src/spark.jobserver/WebApi.scala

```diff
@@ -79,33 +92,36 @@ class WebApi(system: ActorSystem,
    *    POST /jars/<appName>   - upload a new jar file
    */
   def jarRoutes: Route = pathPrefix("jars") {
-    // GET /jars route returns a JSON map of the app name and the last time a jar was uploaded.
-    get { ctx =>
-      val future = (jarManager ? ListJars).mapTo[collection.Map[String, DateTime]]
-      future.map { jarTimeMap =>
-        val stringTimeMap = jarTimeMap.map { case (app, dt) => (app, dt.toString()) }.toMap
-        ctx.complete(stringTimeMap)
-      }.recover {
-        case e: Exception => ctx.complete(500, errMap(e, "ERROR"))
-      }
-    } ~
-      // POST /jars/<appName>
-      // The <appName> needs to be unique; uploading a jar with the same appName will replace it.
-      post {
-        path(Segment) { appName =>
-          entity(as[Array[Byte]]) { jarBytes =>
-            val future = jarManager ? StoreJar(appName, jarBytes)
-            respondWithMediaType(MediaTypes.`application/json`) { ctx =>
-              future.map {
-                case JarStored  => ctx.complete(StatusCodes.OK)
-                case InvalidJar => badRequest(ctx, "Jar is not of the right format")
-              }.recover {
-                case e: Exception => ctx.complete(500, errMap(e, "ERROR"))
+    // user authentication
+    authenticate(authenticator) { authInfo =>
```

Comment:
Correct. Do you prefer an _ instead? We thought to leave it in just in case someone might want to use it at some point of time in the future.